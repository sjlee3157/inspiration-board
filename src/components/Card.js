import React from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './style/Card.css';

const Card = (props) => {
  const displayEmoji = () => {
    const icon = emoji.getUnicode(props.emoji);
    return (icon ? icon : props.emoji)
  };

  const displayDelete = () => {
    return (
      <button className="card__tools-button"
        onClick={ () => props.onDeleteCallback(props.id) }>
        X
      </button>
    )
  }

  const displayEdit = () => {
    const { id, text, emoji } = props;
    return (
      <button className="card__tools-button"
        onClick={ () => props.showEditFormCallback( { id, text, emoji }) }>
        Edit
      </button>
    )
  }

  return (
    <section className="card">
      <div className="card__content">
        <div className="card__content-text">
          { props.children }
          { props.text }
        </div>
        <div className="card__content-emoji">
          { displayEmoji() }
          { /* Todo: place form body here */ }
        </div>
      </div>
      <div className="card__tools">
        { props.id !== Infinity && displayEdit() }
        { props.id !== Infinity && displayDelete() }
      </div>
    </section>
  )
}

Card.propTypes = {
  id: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired,
  emoji: PropTypes.string.isRequired,
  onDeleteCallback: PropTypes.func,
  showEditFormCallback: PropTypes.func
};

export default Card;
