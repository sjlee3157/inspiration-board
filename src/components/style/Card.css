.card {
  background-color: #F4FF81;

  padding: 0;
  margin: 0.5rem;

  min-height: 250px;
  min-width: 225px;
  flex-basis: 18%;

  border-radius: 5px;

  display: grid;
  grid-template-rows: 1fr 4fr 1fr;
  /* align-items: center; */
}

.card__content {
  /* grid-column-start: 2; */
  display: flex;
  flex-direction: column;
  align-self: center;

  text-align: center;
  font-weight: 100;
  overflow: scroll;

  margin: 0 30px;
  order: 2;
}

.card__content-text {
  text-overflow: ellipsis;
  overflow: scroll;
}

.card__content-emoji {
  font-size: 3rem;
}

.card__content-text, .card__content-emoji {
  padding: 0;
  margin: 0;
}

.card__tools {
  /* align-self: start; */
  display: flex;
  justify-content: space-between;
  order: 1;
  margin-left: 0.5em;
}

.card__tools-button {
  background-color: inherit;
  font-family: 'Permanent Marker', Helvetica, sans-serif;
  font-size: 1.2em;
  border: none;
  color: darkgray;
}

.card__tools-button:hover {
  color: black;
}

.card__tools-button:last-of-type {
  margin-left: auto;
}
