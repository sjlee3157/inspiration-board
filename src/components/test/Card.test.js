import React from 'react';
import { shallow } from 'enzyme';
import Card from '../Card';

describe('Card', () => {

  it('will match the Card Snapshot', () => {
    const wrapper = shallow( <Card
      id={ 1 }
      text='succumb sob suffer'
      emoji='poop'
    /> );
  expect(wrapper).toMatchSnapshot();
  });

});
