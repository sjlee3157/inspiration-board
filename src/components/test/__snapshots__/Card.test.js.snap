// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card will match the Card Snapshot 1`] = `
<section
  className="card"
>
  <div
    className="card__content"
  >
    <div
      className="card__content-text"
    >
      succumb sob suffer
    </div>
    <div
      className="card__content-emoji"
    >
      ðŸ’©
    </div>
  </div>
  <div
    className="card__tools"
  >
    <button
      className="card__tools-button"
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      className="card__tools-button"
      onClick={[Function]}
    >
      X
    </button>
  </div>
</section>
`;
